### DISCUSSION TITLE
jans-cli usability

### DISCUSSION THREAD

**Discussion by manojs1978** [JanssenProject/jans#1158] [General]
please find some usability update
[jans-cli_usability.pdf](https://github.com/JanssenProject/jans/files/8409148/jans-cli_usability.pdf)

**Comment by ossdhaval** [comment]
If we can add two more:

`Optional fields` should be sorted or categorised so that options are easy to locate. As shown in image below,  `optional fields` presents a huge list of optional fields. If you are looking for a particular option, it is not easy to find. For example, if it was sorted in alphabetical order, it would be easy. Or may be categorised in some logical way. 

![image](https://user-images.githubusercontent.com/343411/161562217-ca4efad9-8364-4074-baaa-7c6d87633036.png)

Plus, the `Populate optional fields?` doesn't have any suggestions for input.
![image](https://user-images.githubusercontent.com/343411/161561323-fefdf699-68a9-49e5-942b-cf106459d590.png)

**Reply by ossdhaval** [reply]
captured this comment as #1349

**Reply by devrimyatar** [reply]
Categorisation not possible, I will sort them.

**Comment by ossdhaval** [comment]
Prompt for `scopes` optional parameter says `array of string separated by _,»` but the example is space separated

```
«Provide list of scopes granted to the client. Type: array of string separated by _,»
Example: read write dolphin
```

**Reply by ossdhaval** [reply]
Captured this comment as #1350

**Comment by ossdhaval** [comment]
In the end, after finishing giving all inputs, when JSON gets printed on screen, give an option to save it in place of having the user copy it manually.

**Reply by ossdhaval** [reply]
Captured this comment as #1352

**Reply by manojs1978** [reply]
what is expected behaviour if someone not press enter within 1800 sec after device verification or press enter before device verification.?

![image](https://user-images.githubusercontent.com/5729240/168959090-521dcb8a-68be-42b0-ba45-2fd0c81ecaae.png)

i am getting bad request (error 400) but i think it should be timeout if user not press enter within 1800 sec and when pressed enter before device verification it should not give any error till 1800 sec.

**Reply by devrimyatar** [reply]
We can count timeout, an look for the response of the server if device verification is done, if not warn user and wait until user hits enter again.

**Comment by devrimyatar** [comment]
@ossdhaval Thanks for nice suggestions. I will implement these on this branch https://github.com/JanssenProject/jans/tree/jans-cli-no-swagger-client .

**Comment by ossdhaval** [comment]
Closing this as most of the suggestions have been incorporated.
